### PR TITLE
Use the SAL’s new inet_ntop and inet_pton functions

### PR DESCRIPTION
**Pull request by bremoran**
* fmtIPv6 now uses ```inet_ntop(AF_INET6)```
* fmtIPv4 now uses ```inet_ntop(AF_INET)```
* Added a new function: ```setAddr(socket_address_family_t af, char *)```

### CODE REVIEW

**Comment by bremoran** [comment]
cc @bogdanm @niklas-arm @0xc0170 @adbridge

File: sockets/v0/Socket.h [hunk 1]
```diff
@@ -240,19 +240,37 @@ class Socket {
...
+     *  If NULL, error handling is discarded.
+     */
     ErrorHandler_t    _onError;
-    ReadableHandler_t _onReadable;
+    /** Function pointer to the user-supplied data available handling function
+
+     *  This function pointer is called when data is available on the socket. It is called once for
+     *  each event generated by the socket layer, so partial reads can leave data available on the
+     *  socket and greedy reads can cause extra events with no data to read.
+     *  Suggested handling is to use greedy reads and terminate early in events with no */
```
Comment (0xc0170) [inline sockets/v0/Socket.h]: > Suggested handling is to use greedy reads and terminate early in events with no

unfinished sentence? 264 does not need empty line (removal of internal?)

**Comment by 0xc0170** [comment]
Looks good to me. 

This reminds me that we should document SocketAddr API

File: sockets/v0/SocketAddr.h [hunk 2]
```diff
@@ -29,11 +29,12 @@ class SocketAddr {
...
     void setAddr(const struct socket_addr *addr);
     void setAddr(const SocketAddr *addr);
+    int setAddr(socket_address_family_t af, const char *addr);
     size_t getAddrSize() const {return sizeof(_addr.ipv6be);}
-    bool is_v4();
+    bool is_v4() const;
 
-    int fmtIPv4(char *buf, size_t size);
-    int fmtIPv6(char *buf, size_t size);
+    int fmtIPv4(char *buf, size_t size) const;
```
Comment (niklas-arm) [inline sockets/v0/SocketAddr.h]: Wasn't there some more documentation

**Comment by niklas-arm** [comment]
Please update this with the header documentation from #39 !

**Comment by niklas-arm** [comment]
Would you like to add convenience functions like these? They make it a little safer to work with this API.
```
template< size_t N >
inline int fmtIPv4(char (&buf)[N]) {
    return fmtIPv4(buf, N);
}
```

**Comment by bremoran** [comment]
What do I do with the license for this?

I think it would be advisable to take care of the concern that @adbridge brought up above. The switch should not happen twice. It would be better to modify inet_pton to do the right thing.

**Comment by bremoran** [comment]
@niklas-arm said +1 in a separate message. I'm merging this so that other PRs can progress.

**Pull request merged**